lib/dal/src/component.rs

```diff
@@ -1619,12 +1623,81 @@ impl Component {
 
         Ok(serde_json::from_value(type_value)?)
     }
+    /// Sets the [`AttributeValue`] for root/si/type to the given [`ComponentType`]
+    /// NOTE: This does NOT ensure that this change is valid, nor does it account for
+    /// needing to update other attribute values in cases where the new type is an up or
+    /// down frame
+    pub async fn set_type_by_id(
```

Comment:
It's not clear from names that set and update do different things--if this is sort of an "unsafe/unchecked" method, it's probably worth giving one of them a worse name so we can read some sort of difference at callsites. `set_type_by_id_unchecked` vs. `set_type_by_id` maybe, so people using autocomplete hesitate before calling the unchecked one? I think you could call this a nit, but it confused me quickly seeing the two different calls in the PR right away :)

(Could be we have a convention for what "update" means I'm not aware of though!